tests/blocks/test_notifications.py

```diff
@@ -1,54 +1,86 @@
-from unittest.mock import MagicMock
+from importlib import reload
+from typing import Type
+from unittest.mock import patch
 
 import cloudpickle
+import pytest
 
-from prefect import flow
-from prefect.blocks.notifications import SlackWebhook
+import prefect
+from prefect.blocks.notifications import AppriseNotificationBlock, PrefectNotifyType
 from prefect.testing.utilities import AsyncMock
 
 
-class TestSlackWebhook:
-    async def test_notify_async(self, monkeypatch):
-        async_webhook_client_mock = AsyncMock()
-        async_webhook_client_constructor_mock = MagicMock(
-            return_value=async_webhook_client_mock
-        )
-        monkeypatch.setattr(
-            "slack_sdk.webhook.async_client.AsyncWebhookClient",
-            async_webhook_client_constructor_mock,
-        )
-
-        block = SlackWebhook(url="http://example.com/slack")
-        await block.notify("test")
-
-        async_webhook_client_constructor_mock.assert_called_once_with(
-            url=block.url.get_secret_value()
-        )
-        async_webhook_client_mock.send.assert_called_once_with(text="test")
-
-    def test_notify_sync(self, monkeypatch):
-        async_webhook_client_mock = AsyncMock()
-        async_webhook_client_constructor_mock = MagicMock(
-            return_value=async_webhook_client_mock
-        )
-        monkeypatch.setattr(
-            "slack_sdk.webhook.async_client.AsyncWebhookClient",
-            async_webhook_client_constructor_mock,
-        )
-
-        url = "http://example.com/slack"
-
-        @flow
-        def test_flow():
-            block = SlackWebhook(url=url)
-            block.notify("test")
-
-        test_flow()
-        async_webhook_client_constructor_mock.assert_called_once_with(url=url)
-        async_webhook_client_mock.send.assert_called_once_with(text="test")
-
-    def test_is_picklable(self):
-        block = SlackWebhook(url="http://example.com/slack")
+def reload_modules():
+    """
+    Reloads the prefect.blocks.notifications module so patches to modules it imports
+    will be visible to the blocks under test.
+    """
+    try:
+        reload(prefect.blocks.notifications)
+    except UserWarning as ex:
+        # ignore the warning Prefect gives when reloading the notifications module
+        # because we reload prefect itself immediately afterward.
+        pass
+
+    reload(prefect)
+
+
+# A list of the notification classes Pytest should use as parameters to each method in TestAppriseNotificationBlock
+notification_classes = AppriseNotificationBlock.__subclasses__()
```

Comment:
It might be better to explicitly list the classes we want to test instead of including all subclasses. On the other hand, including them all lets us use this test class as a way of validating expected behavior for any new Apprise-based notification blocks we add.

Reply:
I'm a fan of this approach. I like that it standardizes the behavior that we should expect from all Apprise notification blocks and does require either the implementer or review to remember to add a new Apprise notification block to a list somewhere. 

I think that when parameterizing tests, a deterministic order is important because the different test runs are distributed across different workers in the CI pipeline. I'd recommend adding a sort function so that `notification_classes` is always in the same order for each execution of this module.

Reply:
Thanks for the great suggestion! `notification classes` is now sorted.